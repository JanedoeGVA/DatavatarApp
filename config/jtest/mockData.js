const logoStrava = require('../../src/assets/images/strava-logo.png');
const logoGarmin = require('../../src/assets/images/garmin-logo.png');

const trackers = [
  {
    id: 10,
    provider: 'Strava',
    isAvailable: true,
    protocol: '',
    token: {
      accessTokenKey: '',
      refreshTokenKey: '',
      accessTokenSecret: ''
    },
    logo: logoStrava
  },
  {
    id: 11,
    provider: 'Garmin',
    isAvailable: true,
    protocol: '',
    token: {
      accessTokenKey: '',
      refreshTokenKey: '',
      accessTokenSecret: ''
    },

    logo: logoGarmin
  }
];

export default trackers;
