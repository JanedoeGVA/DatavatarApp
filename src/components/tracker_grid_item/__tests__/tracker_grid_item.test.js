import React from 'react';
import renderer from 'react-test-renderer';
import TrackerGridItem from '../../tracker_grid_item';
import { trackers } from '../../../../config/jtest/mockData';

it('renders a TrackerGridItem using Snapshots', () => {
  expect(
    renderer.create(
      <TrackerGridItem
        onPressItem={jest.fn}
        setItemColor={jest.fn}
        item={trackers[0]}
      />
    )
  ).toMatchSnapshot();
});
