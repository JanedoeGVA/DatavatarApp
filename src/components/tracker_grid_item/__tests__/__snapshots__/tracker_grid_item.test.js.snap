// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tracker Grid Item Component renders a TrackerGridItem using Snapshots 1`] = `
<View
  accessible={true}
  isTVSelectable={true}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={null}
>
  <View
    style={
      Array [
        Object {
          "borderRadius": 5,
          "height": 150,
          "justifyContent": "flex-end",
          "padding": 10,
        },
        Object {
          "backgroundColor": [MockFunction],
        },
      ]
    }
  >
    <Image
      activeOpacity={50}
      source={
        Object {
          "testUri": "../../../src/assets/images/strava-logo.png",
        }
      }
      style={
        Object {
          "alignSelf": "center",
          "flex": 1,
          "height": 80,
          "resizeMode": "contain",
          "width": 80,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "600",
        }
      }
    >
      true
    </Text>
  </View>
</View>
`;
